Week3/homework/step2-4.js

```diff
@@ -1,10 +1,12 @@
 'use strict';
 
-function Dog() {
-  // add your code here
+function Dog(name, color) {
+  this.name = name;
+  this.color = color;
+  this.numLegs = 4;
```

Comment:
Aahaan. This isn't the best way of creating a constructor function. This is because constructor functions are like a factory which can be used x number of times to create objects(dogs in this case). Thus, if we use the existing constructor function, we would get the same dog everytime while we wanted different ones. A better way to do this would be to use function parameters/arguments.

```
function Dog(name, color, legs) {
  this.name = name;
  this.color = color;
  this.numLegs = legs;
}

const dog1 = new Dog('Georgie', 'White', 4);
const dog2 = new Dog('Lenny', 'Brown', 4);
// and so on
```

Reply:
Now with your explanation I clearly see that my code would not allowing modify number of legs :) You're right, I am changing it